src/Microsoft.ML.Transforms/CategoricalTransform.cs

```diff
@@ -212,15 +214,17 @@ internal void SetTerms(string terms)
         /// <param name="outputKind">The type of output expected.</param>
         public CategoricalEstimator(IHostEnvironment env, string name,
             string source = null, CategoricalTransform.OutputKind outputKind = Defaults.OutKind)
-            : this(env, new ColumnInfo(source ?? name, name, outputKind))
+            : this(env, new[] { new ColumnInfo(source ?? name, name, outputKind) })
         {
         }
 
-        public CategoricalEstimator(IHostEnvironment env, params ColumnInfo[] columns)
+        public CategoricalEstimator(IHostEnvironment env, ColumnInfo[] columns,
```

Comment:
>public [](start = 8, length = 6)

out of curiosity can you make this constructor internal? I would prefer to not pollute our public API with these things.
Same for TermEstimator.